packages/reactive-element/src/decorators/listen.ts

```diff
@@ -0,0 +1,165 @@
+ * not an arrow function.
+ */
+
+import {ReactiveElement} from '../reactive-element.js';
+import {decorateProperty} from './base.js';
+
+interface ListenInfo {
+  type: string;
+  prop: PropertyKey;
+  options?: EventListenerOptions;
+  target: '' | 'root' | EventTarget;
+}
+
+type Listeners = ListenInfo[];
+
+const hostListeners: WeakMap<typeof ReactiveElement, Listeners> = new WeakMap();
+const rootListeners: WeakMap<typeof ReactiveElement, Listeners> = new WeakMap();
+const targetedListeners: WeakMap<
+  typeof ReactiveElement,
+  Listeners
+> = new WeakMap();
+
+// This is done so we can call the listener in the right context.
+// TODO(sorvell): Alternatively, can require users to bind the function, but this
+// seemslike a foot gun.
+const listenerMap: WeakMap<
+  ReactiveElement,
+  Map<ListenInfo, EventListener>
+> = new WeakMap();
+
+const getListener = (
+  el: ReactiveElement,
+  listener: EventListener,
+  info: ListenInfo
+) => {
+  let listeners = listenerMap.get(el);
+  if (listeners === undefined) {
+    listenerMap.set(el, (listeners = new Map()));
+  }
+  let listenFn = listeners.get(info);
+  if (listenFn === undefined) {
+    listeners.set(info, (listenFn = listener.bind(el)));
+  }
+  return listenFn;
+};
+
+const addRemoveListeners = (
+  el: ReactiveElement,
+  listeners: Listeners,
+  remove = false
+) => {
+  listeners.forEach((info: ListenInfo) => {
+    const {type, prop, options, target} = info;
+    const eventTarget =
+      target === 'root'
+        ? el.renderRoot
+        : target instanceof EventTarget
+        ? target
+        : el;
+    const listener = (((el as unknown) as {[prop: string]: Element | null})[
+      prop as string
+    ] as unknown) as EventListener;
+    const listenFn = getListener(el, listener, info);
+    if (remove) {
+      eventTarget.removeEventListener(type, listenFn, options);
+    } else {
+      eventTarget.addEventListener(type, listenFn, options);
+    }
+  });
+};
+
+const hasRootListeners: Set<ReactiveElement> = new Set();
+
+/**
+ * Adds an event listener.
+ *
+ * @param options {object} An object specifying event listener options. This
+ * object must include the `type` of the event for which to listen and
+ * optionally may include an `options` property to specify the
+ * `EventListenerOptions` and a `target` property. When `target` is not provided
+ * the event listener is installed on the element itself. If set to `root`,
+ * the listener is installed on the element `renderRoot`. Otherwise
+ * `target` is assumed to be an `EventTarget` and the listener is installed
+ * directly on it. When the `target` is an `EventTarget`, the listener is
+ * dynamically added and removed when the element is connected/disconnected
+ * from the DOM.
+ *
+ * @example
+ * ```ts
+ * class MyElement {
+ *   @listen('click')
+ *   private _clickHandler(e: Event) => {
+ *     this.clickedOn = e.target.localName;
+ *   }
+ *   @listen('keydown', window, {capture: true}))
```

Comment:
I noticed that the docs do not match the function definition. 

`listen('keydown', window, { capture: true })` 
vs 
`listen({ type: 'keydown', options: { capture: true }, target: window })`